[Style] Convert the 'position-area' property to strong style types

weinig opened the pull request:
#### 0e3e841bc32d4cd96c3bb13421d151aae58bc0c3
<pre>
[Style] Convert the &apos;position-area&apos; property to strong style types
<a href="https://bugs.webkit.org/show_bug.cgi?id=302727">https://bugs.webkit.org/show_bug.cgi?id=302727</a>

Reviewed by Darin Adler.

Converts the &apos;position-area&apos; property to use strong style types.

Also splits up the types previously in PositionArea.h/cpp into files
of their own.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
* Source/WebCore/rendering/RenderInline.cpp:
* Source/WebCore/rendering/style/PositionArea.cpp: Removed.
* Source/WebCore/rendering/style/PositionArea.h: Removed.
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderState.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/values/anchor-position/StylePositionArea.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionArea.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSelf.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSelf.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSpan.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSpan.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaTrack.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaTrack.h: Added.

Canonical link: <a href="https://commits.webkit.org/303452@main">https://commits.webkit.org/303452@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e6aedfde5d9298dd46dc3b341ada9a39062b7ed8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132457 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4952 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43514 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139972 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84424 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134327 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5019 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4711 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101266 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68514 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135403 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3537 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118659 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82059 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1242 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83203 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112281 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36776 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142626 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4622 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37365 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109641 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4704 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3990 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109822 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27827 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3505 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114933 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57972 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4676 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33283 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4510 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68127 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4767 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4633 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionArea.h:
```diff
@@ -0,0 +1,102 @@
+    PositionAreaValue(PositionAreaSpan blockOrXAxis, PositionAreaSpan inlineOrYAxis);
+
+    PositionAreaSpan blockOrXAxis() const { return m_blockOrXAxis; }
+    PositionAreaSpan inlineOrYAxis() const { return m_inlineOrYAxis; }
```
These could be `constexpr`.

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionArea.h:
```diff
@@ -0,0 +1,102 @@
+
+    bool operator==(const PositionAreaValue&) const = default;
+
+private:
+    PositionAreaSpan m_blockOrXAxis;
+    PositionAreaSpan m_inlineOrYAxis;
```
Not sure why we have private data members in a `struct`. Maybe I’ve been missing the pattern we are following everywhere. Shouldn’t we be using public data members more?

weinig replied:
Yeah, this was a holdover, but these can be public.

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h:
```diff
@@ -0,0 +1,107 @@
+    return PositionAreaAxis::Horizontal;
+}
+
+inline bool isPositionAreaAxisLogical(const PositionAreaAxis positionAxis)
```
```suggestion
constexpr bool isPositionAreaAxisLogical(PositionAreaAxis positionAxis)
```

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h:
```diff
@@ -0,0 +1,107 @@
+    return static_cast<uint8_t>(positionAxis) & axisBit;
+}
+
+inline bool isPositionAreaDirectionLogical(const PositionAreaAxis positionAxis)
```
```suggestion
constexpr bool isPositionAreaDirectionLogical(PositionAreaAxis positionAxis)
```

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h:
```diff
@@ -0,0 +1,107 @@
+    return static_cast<uint8_t>(positionAxis) & directionBit;
+}
+
+inline BoxAxis mapPositionAreaAxisToPhysicalAxis(const PositionAreaAxis positionAxis, const WritingMode writingMode)
```
```suggestion
constexpr BoxAxis mapPositionAreaAxisToPhysicalAxis(PositionAreaAxis positionAxis, WritingMode writingMode)
```

darinadler commented on Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h:
```diff
@@ -0,0 +1,107 @@
+    return static_cast<BoxAxis>(physicalAxis);
+}
+
+inline LogicalBoxAxis mapPositionAreaAxisToLogicalAxis(const PositionAreaAxis positionAxis, const WritingMode writingMode)
```
```suggestion
constexpr LogicalBoxAxis mapPositionAreaAxisToLogicalAxis(PositionAreaAxis positionAxis, WritingMode writingMode)
```

weinig committed:
[Style] Convert the 'position-area' property to strong style types
https://bugs.webkit.org/show_bug.cgi?id=302727

Reviewed by Darin Adler.

Converts the 'position-area' property to use strong style types.

Also splits up the types previously in PositionArea.h/cpp into files
of their own.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
* Source/WebCore/rendering/RenderInline.cpp:
* Source/WebCore/rendering/style/PositionArea.cpp: Removed.
* Source/WebCore/rendering/style/PositionArea.h: Removed.
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderState.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/values/anchor-position/StylePositionArea.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionArea.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaAxis.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSelf.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSelf.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSpan.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaSpan.h: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaTrack.cpp: Added.
* Source/WebCore/style/values/anchor-position/StylePositionAreaTrack.h: Added.

Canonical link: https://commits.webkit.org/303452@main
